### PR TITLE
[Fix] RayEngine multi-node: co-locate rank0 scheduler with Engine and fix CUDA device setting

### PR DESCRIPTION
**Pull request by xyuzh**
## Problem
Multi-node serving fails because Engine got initialized at a different node from the rank0 scheduler, this causes the sglang `Engine` to have indeterministic dangled `generate` due to the scheduler with rank0 running the event loop fail to read request from TokenizerManager through ZMQ communication and this causes hanging.

## Summary
- **Co-locate rank0 scheduler(tp_rank=0, pp_rank=0) with Engine node** in the multi-node setting: Replace `_get_rank0_node_ip` with `_find_engine_bundle` that discovers which placement group bundle is on the same node as the Engine. PP0 TP0 communicates with TokenizerManager/Detokenizer via IPC ZMQ, so it must be co-located.

- **Restore CUDA device in `run_event_loop`**: Added `torch.cuda.set_device(self.scheduler.gpu_id)` before entering the event loop in `SchedulerActor`, ensuring NCCL operates on the correct GPU.

## Test plan
- [x] Multi-node Ray serving with PP > 1 (verifies PP0 lands on the Engine node)

The fix would allow ray serve to run in multi-node cluster:
```
@serve.deployment(
    autoscaling_config={
        "min_replicas": 1,
        "max_replicas": 4,
        "target_ongoing_requests": 4,
    },
    ray_actor_options={
        "num_cpus": 1,
        "num_gpus": 0,
    },
    # Reserve resources across multiple nodes for tensor parallelism.
    # Each bundle reserves GPUs on one node.
    placement_group_bundles=[{"CPU": 1, "GPU": gpus_per_node}] * NUM_NODES,
)
@serve.ingress(app)
class SGLangDeployment:
    def __init__(self):
        from sglang.srt.ray.engine import RayEngine as Engine

        logger.info(f"Initializing engine with model path: {MODEL_PATH}")
        self.engine = Engine(
            model_path=MODEL_PATH,
            tp_size=TP_SIZE,
            pp_size=PP_SIZE,
            nnodes=NUM_NODES,
        )

    @app.post("/")
    async def generate(self, request: dict) -> dict:
        result = await self.engine.async_generate(
            prompt=text,
        )
        return {"text": result["text"]}


app_deploy = SGLangDeployment.bind()
```

### CODE REVIEW

File: python/sglang/srt/ray/scheduler_actor.py [hunk 1]
```diff
@@ -105,6 +105,10 @@ def get_info(self) -> Dict[str, Any]:
     def run_event_loop(self) -> None:
         """Run the scheduler's event loop. Blocks until shutdown."""
         try:
+            import torch
+
+            # Need to set the GPU id for the event loop for nccl to work
+            torch.cuda.set_device(self.scheduler.gpu_id)
```
Comment (xyuzh) [inline python/sglang/srt/ray/scheduler_actor.py:111]: Ray actors can run each remote method call on a different thread. CUDA device context is thread-local in PyTorch. So even though __init__ calls torch.cuda.set_device (via init_torch_distributed in the ModelRunner), when Ray later invokes run_event_loop() as a separate remote call, it may execute on a different thread where no CUDA device has been set.

File: python/sglang/srt/ray/engine.py [hunk 2]
```diff
@@ -41,27 +41,37 @@ class RaySchedulerInitResult(SchedulerInitResult):
...
+                ),
+            ).remote()
+            for i in range(nnodes)
+        ]
     )
 
+    try:
+        return bundle_ips.index(engine_ip), engine_ip
+    except ValueError:
+        logger.warning(
```
Comment (Qiaolin-Yu) [inline python/sglang/srt/ray/engine.py]: Should we just raise error here and make it crash? Will this fallback path make it harder to debug?

**Review by Qiaolin-Yu — COMMENTED** [review]
overall lgtm. left a nit

File: python/sglang/srt/ray/engine.py (hunk 2 above)
Comment (xyuzh) [reply python/sglang/srt/ray/engine.py]: I think crash is ok, if engine ip is not found in the ray cluster nodes, it should be an error, I updated the handling to fail directly

**Review by Qiaolin-Yu — APPROVED** [review]
lgtm
